crates/providers/http-client/src/lib.rs

```diff
@@ -1,177 +1,117 @@
-use std::collections::HashMap;
-use std::str::FromStr;
+use core::future::Future;
+use core::pin::pin;
 
 use anyhow::Context as _;
-use anyhow::Result;
-use http::{HeaderMap, HeaderName, HeaderValue};
-use tracing::{error, instrument, trace, warn};
+use futures::StreamExt;
+use hyper_util::rt::TokioExecutor;
+use tokio::{select, spawn};
+use tracing::{debug, error, instrument, warn};
+use wasmcloud_provider_sdk::{get_connection, ProviderHandler};
+use wrpc_interface_http::try_http_to_outgoing_response;
+use wrpc_transport::AcceptedInvocation;
 
-use wasmcloud_provider_wit_bindgen::deps::{
-    async_trait::async_trait, wasmcloud_provider_sdk::core::LinkDefinition,
-    wasmcloud_provider_sdk::Context,
-};
-
-wasmcloud_provider_wit_bindgen::generate!({
-    impl_struct: HttpClientProvider,
-    contract: "wasmcloud:httpclient",
-    replace_witified_maps: true,
-    wit_bindgen_cfg: "provider-http-client"
-});
+#[instrument(level = "trace", skip_all)]
+pub async fn serve_handle<Ctx, Tx, C>(
+    client: hyper_util::client::legacy::Client<
```

Comment:
Do we have to worry about using the legacy client here? Just want to make sure it isn't going to get ripped out soon

Reply:
not really - there is no other client currently provided by hyper and is what's suggested to use in the docs of https://docs.rs/hyper-rustls/latest/hyper_rustls/index.html